### PR TITLE
[FX-4605] Migrate AccordionDetails to TailwindCSS

### PR DESCRIPTION
**Pull request by AdrianContiu**
[FX-4605]

### Description

This PR removes material-ui styles and replace them with Tailwind CSS classes.

### How to test

- [Temploy](https://picasso.toptal.net/fx-4605-migrate-accordiondetails)
- Check the [production](https://picasso.toptal.net/?path=/story/components-accordion--accordion) and compare it with the temploy.

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| image | Insert screenshots or screen recordings |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- ~Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)~
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- ~Annotate all `props` in component with documentation~
- ~Create `examples` for component~
- ~Ensure that deployed demo has expected results and good examples~
- ~Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).~
- [x] Self reviewed
- ~Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)~

**Breaking change**

- ~codemod is created and showcased in the changeset~
- ~test alpha package of Picasso in StaffPortal~

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4605]: https://toptal-core.atlassian.net/browse/FX-4605?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/base/Accordion/src/AccordionDetails/AccordionDetails.tsx [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
-
-const AccordionDetails = (props: Props) => {
-  const {
-    children,
-    // Avoid passing external classes inside the rest props
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    classes: externalClasses,
-    ...rest
-  } = props
+const AccordionDetails = forwardRef<HTMLDivElement, Props>(
```
Comment (augustobmoura) [inline packages/base/Accordion/src/AccordionDetails/AccordionDetails.tsx:10]: What is the reason for wrapping this component on `forwardRef` on this PR? If not necessary I think we should avoid it. We weren't forwarding `ref`s before this

File: packages/base/Accordion/src/AccordionDetails/AccordionDetails.tsx [hunk 2]
```diff
@@ -1,36 +1,29 @@
...
-  const classes = useStyles(props)
-
-  return (
-    <MUIAccordionDetails {...rest} classes={classes}>
-      {children}
-    </MUIAccordionDetails>
-  )
-}
+    return (
+      <div {...rest} className={twMerge('flex p-0', className)} ref={ref}>
```
Comment (augustobmoura) [inline packages/base/Accordion/src/AccordionDetails/AccordionDetails.tsx:22]: We didn't support custom `className` before, do we need to support it? Maybe we can simplify the code by avoiding `twMerge` while also not exposing a new API for users to depend on support. Remember [Hyrum's Law](https://www.hyrumslaw.com/) and [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter). If we can make our APIs simpler and provide only the necessary, I think we should

File: packages/base/Accordion/src/AccordionDetails/AccordionDetails.tsx (hunk 1 above)
Comment (AdrianContiu) [reply packages/base/Accordion/src/AccordionDetails/AccordionDetails.tsx]: I got confused by following the [implementation](https://github.com/mui/material-ui/blob/v4.x/packages/material-ui/src/AccordionDetails/AccordionDetails.js) of `AccordionDetails` on material-ui. I have removed everything as per your recommendations.

image

File: packages/base/Accordion/src/AccordionDetails/AccordionDetails.tsx (hunk 2 above)
Comment (AdrianContiu) [reply packages/base/Accordion/src/AccordionDetails/AccordionDetails.tsx]: Similar to my first comment. However, thank you for sharing these important concepts!

**Review by TomasSlama — APPROVED** [review]

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

**Review by augustobmoura — APPROVED** [review]
Nice work @AdrianContiu! :+1:

**Pull request merged**